airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/GenerateInputActivityImpl.java

```diff
@@ -128,25 +179,27 @@ public GeneratedJobInput getSyncWorkflowInput(final SyncInput input) {
           .withNormalizationDockerImage(destinationNormalizationDockerImage)
           .withSupportsDbt(destinationDefinition.getSupportsDbt())
           .withNormalizationIntegrationType(normalizationIntegrationType)
-          .withAllowedHosts(configReplacer.getAllowedHosts(destinationDefinition.getAllowedHosts(), config.getDestinationConfiguration()));
+          .withAllowedHosts(configReplacer.getAllowedHosts(destinationDefinition.getAllowedHosts(), attemptSyncConfig.getDestinationConfiguration()));
 
       final StandardSyncInput syncInput = new StandardSyncInput()
           .withNamespaceDefinition(config.getNamespaceDefinition())
           .withNamespaceFormat(config.getNamespaceFormat())
           .withPrefix(config.getPrefix())
           .withSourceId(standardSync.getSourceId())
           .withDestinationId(standardSync.getDestinationId())
-          .withSourceConfiguration(config.getSourceConfiguration())
-          .withDestinationConfiguration(config.getDestinationConfiguration())
+          .withSourceConfiguration(attemptSyncConfig.getSourceConfiguration())
+          .withDestinationConfiguration(attemptSyncConfig.getDestinationConfiguration())
           .withOperationSequence(config.getOperationSequence())
           .withWebhookOperationConfigs(config.getWebhookOperationConfigs())
           .withCatalog(config.getConfiguredAirbyteCatalog())
-          .withState(config.getState())
+          .withState(attemptSyncConfig.getState())
           .withResourceRequirements(config.getResourceRequirements())
           .withSourceResourceRequirements(config.getSourceResourceRequirements())
           .withDestinationResourceRequirements(config.getDestinationResourceRequirements())
           .withWorkspaceId(config.getWorkspaceId());
 
+      saveAttemptSyncConfig(jobId, attempt, connectionId, attemptSyncConfig);
```

Comment:
After generating the input, we persist the actor configuration and state used. This is done via an internal api call.